hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/MetricsRegionServer.java

```diff
@@ -122,89 +101,79 @@ public MetricsRegionServerWrapper getRegionServerWrapper() {
     return regionServerWrapper;
   }
 
-  public void updatePutBatch(TableName tn, long t) {
-    if (tableMetrics != null && tn != null) {
-      tableMetrics.updatePutBatch(tn, t);
-    }
+  public void updatePutBatch(HRegion region, long t) {
+    Optional.ofNullable(region).ifPresent(r -> Optional.ofNullable(r.getMetricsTableRequests())
```

Comment:
Oh, I thought you were going to remove all the Optional stuff. 

If you want to keep that, I think we should just have region.getTableMetrics() return an Optional. That way we don't have to create 2 optionals in every call. You can an just chain them together with flatMap.

Reply:
Oh yes, my fault...